mm2src/coins/hd_wallet/mod.rs

```diff
@@ -241,20 +242,61 @@ where
     }
 }
 
-pub async fn load_hd_accounts_from_storage<HDAddress, ExtendedPublicKey>(
+pub async fn load_hd_accounts_from_storage_with_matching_xpubs<HDAddress>(
     hd_wallet_storage: &HDWalletCoinStorage,
     derivation_path: &HDPathToCoin,
-) -> HDWalletStorageResult<HDAccountsMap<HDAccount<HDAddress, ExtendedPublicKey>>>
+    global_hd_ctx: &GlobalHDAccountArc,
+) -> HDWalletStorageResult<HDAccountsMap<HDAccount<HDAddress, ExtendedPublicKey<secp256k1::PublicKey>>>>
 where
     HDAddress: HDAddressOps + Send,
-    ExtendedPublicKey: ExtendedPublicKeyOps,
-    <ExtendedPublicKey as FromStr>::Err: Display,
 {
+    let coin_der_path = derivation_path.to_derivation_path();
     let accounts = hd_wallet_storage.load_all_accounts().await?;
-    let res: HDWalletStorageResult<HDAccountsMap<HDAccount<HDAddress, ExtendedPublicKey>>> = accounts
-        .iter()
+
+    // Filter accounts that when derived from the current HD wallet, they yield the correct xpub.
+    // If they don't yield the correct xpub, this means that account belong to another HD wallet with
+    // different purpose'/coin_type' fields but same account_id' field. This happens when someone changes
+    // the purpose'/coin_type' fields in the coins cnofig without changing the ticker.
+    // TODO: This is a temporary solution that fixes the issue. But we might wanna re-design how accounts
+    //       are loaded from the database (loading only the single account the user asked for and not all accounts).
+    // FIXME: What about collecting unmatching xpubs and deleting them. So that they never clutter the space of
```

Comment:
I agree, if we change a derivation path for a ticker we should delete the old entries

Reply:
[062a268](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2482/commits/062a268b25f17f4f7ca0f57ea9f0b1b028aaecea)